packages/SwingSet/src/liveslots/virtualReferences.js

```diff
@@ -160,6 +165,46 @@ export function makeVirtualReferenceManager(
     kindInfoTable.set(`${kindID}`, { reanimator, deleter, durable });
   }
 
+  /**
+   * Compare two arrays (shallowly) for equality.
+   *
+   * @template T
+   * @param {T[]} a1
+   * @param {T[]} a2
+   * @returns {boolean}
+   */
+  function arrayEquals(a1, a2) {
+    assert(Array.isArray(a1));
+    assert(Array.isArray(a2));
+    if (a1.length !== a2.length) {
+      return false;
+    }
+    return a1.every((elem, idx) => a2[idx] === elem);
```

Comment:
I see that you're only using it for an array of strings, for which it is correct, and you're not exporting it, so no immediate dangers of this being used for anything else. But you're typing it as if it could be used for non-strings, in which case it should be correct for non-strings. Your code will fail on `arrayEquals([NaN], [NaN])`.
```suggestion
    return a1.every((elem, idx) => Object.is(a2[idx], elem));
```